File: app/src/main/java/com/example/xoxoxo/MainActivity.java
package com.example.xoxoxo;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;


public class MainActivity extends AppCompatActivity {
    ImageView imageGame_01, imageGame_02;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        imageGame_01 = findViewById(R.id.icon_game_01);
        imageGame_02 = findViewById(R.id.icon_game_02);
        }

    public void startGame_01(View view) {
        Intent intent = new Intent(MainActivity.this, xoxo.class);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/com/example/fragment/MainActivity.kt
package com.example.fragment

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentTransaction

class MainActivity : AppCompatActivity() {

    private lateinit var fragmentTextView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        fragmentTextView = findViewById(R.id.fragmentNameTextView)

        val buttonFragment1: Button = findViewById(R.id.buttonFragment1)
        val buttonFragment2: Button = findViewById(R.id.buttonFragment2)

        buttonFragment1.setOnClickListener {
            loadFragment(FirstFragment())
        }

        buttonFragment2.setOnClickListener {
            loadFragment(SecondFragment())
        }

        Toast.makeText(this, "onCreate called", Toast.LENGTH_SHORT).show()
    }

    private fun loadFragment(fragment: Fragment) {
        val fragmentTransaction: FragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.fragmentContainer, fragment)
        fragmentTransaction.addToBackStack(null)
        fragmentTransaction.commit()
    }

    override fun onStart() {
        super.onStart()
        Toast.makeText(this, "onStart called", Toast.LENGTH_SHORT).show()
    }

    override fun onResume() {
        super.onResume()
        Toast.makeText(this, "onResume called", Toast.LENGTH_SHORT).show()
    }

    override fun onPause() {
        super.onPause()
        Toast.makeText(this, "onPause called", Toast.LENGTH_SHORT).show()
    }

    override fun onStop() {
        super.onStop()
        Toast.makeText(this, "onStop called", Toast.LENGTH_SHORT).show()
    }

    override fun onDestroy() {
        super.onDestroy()
        Toast.makeText(this, "onDestroy called", Toast.LENGTH_SHORT).show()
    }
}
